docs/feature_pointing_device.md

```diff
@@ -731,3 +734,606 @@ In general the following two functions must be implemented in appropriate locati
+
+### Basic Custom Modes
+There are a couple of ways to add new pointing device modes, using the pointing device mode maps will be covered here under basic use where the other method of using the `process_pointing_mode_*` callbacks will be covered under advance use.
+
+#### Pointing Device Mode Maps
+The easiest way to add pointing device modes that are only using keycode taps (similar to `PM_CARET`, `PM_VOLUME`, and `PM_HISTORY`) are through creating pointing device mode maps.
+
+```c
+// Pointing Device Mode Maps Format
+const uint16_t PROGMEM pointing_device_mode_maps[][4] = {
+    [0] = POINTING_MODE_LAYOUT(
+                <keycode up>,
+        <keycode left>,       <keycode right>,
+                <keycode down>
+    ),
+    //... all other pointing mode maps ...
+    [<POINTING_MODE_MAP_COUNT - 1>] = POINTING_MODE_LAYOUT(
+                <keycode up>,
+        <keycode left>,       <keycode right>,
+                <keycode down>
+    )
+}
+
+```
+
+#### Example Mode Maps:
+```c
+// in config.h:
+#define POINTING_DEVICE_MODES_ENABLE // (Required)
+#define POINTING_MODE_MAP_COUNT 4    // (Required) number of modes in map
+#define EXTRAKEY_ENABLE              // (optional) matters for mode id numbers
+// pointing mode map start is left at default value in this example
+
+// in keymap.c
+// required enum to set mode id's
+enum keymap_pointing_mode_ids {
+    PM_BROW = POINTING_MODE_MAP_START,  // BROWSER TAB Manipulation                           [mode id  6]
+    PM_RGB_MODE_VAL,                    // RGB Control for mode and Brightness                [mode id  7]
+    PM_RGB_HUE_SAT,                     // RGB Control for HUE and Saturation                 [mode id  8]
+    PM_RGB_SPEED,                       // RGB Control for Speed                              [mode id  9]
+    POSTMAP_PM_SAFE_RANGE               // To avoid overlap when adding advanced custom modes [mode id 10]
+};
+// (optional) enum to make things easier to read (index numbers can be used directly)
+// Must be in the same order as the above modes
+enum keymap_pointing_mode_maps_index {
+    _PM_BROW,          // index 0
+    _PM_RGB_MODE_VAL,  // index 1
+    _PM_RGB_HUE_SAT,   // index 2
+    _PM_RGB_SPEED      // index 3
+};
+
+const uint16_t PROGMEM pointing_device_mode_maps[][4] = {
+    [_PM_BROW] = POINTING_MODE_LAYOUT(
+                C(S(KC_PGUP)),
+        C(S(KC_TAB)),       C(KC_TAB),
+                C(S(KC_PGDN))
+    ),
+    [_PM_RGB_MODE_VAL] = POINTING_MODE_LAYOUT(
+                RGB_VAI,
+        RGB_RMOD,        RGB_MOD,
+                RGB_VAD
+    ),
+    [_PM_RGB_HUE_SAT] = POINTING_MODE_LAYOUT(
+                RGB_SAI,
+        RGB_HUD,        RGB_HUI,
+                RGB_SAD
+    ),
+    [_PM_RGB_SPEED] = POINTING_MODE_LAYOUT(
+                KC_NO,
+        RGB_SPD,        RGB_SPI,
+                KC_NO
+    )
+};
```

Comment:
Oh drat I should have reviewed tap_code16's code more carefully. I just assumed any 16 bit code would work. (I currently do not have a keyboard with RGB led's to test with)

Will need to review if there is a way to allow for this and just cut it from the docs if not.

Reply:
there are a few ways to do this, but mostly, it would need to call action_exec to best handle this.  Combos and encoder map does this, for instance.

Reply:
Thanks again Drashna I'll quickly review how encoder maps/Combos handle this and based on the complexity to implement  and how much it might delay an update I'll either add it now or consider it for a future PR.

Reply:
Okay I now see that there are a few ways to skin this cat as you mentioned however most of them involve some sort of major downside unfortunately...

The most realistic solution as I understand it would require:
1. Making use of the 16bit keycode variable added to `keyevent_t` by combos (or adding another if necessary)
2. Reserve another remote matrix position to allow processing of the keyevent in `keyboard.h`
3. Set up macros for the new keyevent type
4. Ensure that this doesn't collide with combos use of the `keyevent.keycode` variable (I think this is mostly handled by step 2)
5. Set up the code to call `process_record` on the faux keyevent (handling both press and release)
6. Replace `tap_code_delay16` with the above

Overall not a huge amount of work, but involves touching a lot of QMK code I'll review adding this after I handle the rest of the mentioned issues with the next update.